### ISSUE TITLE
✍️ Contribution period: Saisri Vishwanath

### ISSUE THREAD

**Issue by saisri0102** [ersilia-os/ersilia#1047]
### Week 1 - Get to know the community

- [X] Join the communication channels
- [X] Open a GitHub issue (this one!)
- [X] Install the Ersilia Model Hub and test the simplest model
- [X] Install Docker if needed, and test another model
- [x] Write a motivation statement to work at Ersilia
- [x] Submit your first contribution to the Outreachy site

### Week 2 - Get Familiar with Machine Learning for Chemistry

- [x] Select a model from the list suggested in GitBook
- [x] Download and serve the model via the Ersilia Model Hub to ensure it works
- [x] Open a repository on your GitHub user with all the necessary files
- [x] Select and clean a dataset of 1000 molecules (example notebook 1)
- [x] Run predictions for the molecules on the selected model and evaluate the results

### Week 3 - Validate a Model in the Wild

- [ ] Find a suitable dataset with sufficient experimental results
- [ ] Clean and standardize the dataset
- [ ] Run predictions and calculate metrics.

### Week 4 - Prepare your final application

- [ ] Submit the final application in the Outreachy website

**Comment by saisri0102** [comment]
Hi,

I am writing to express my strong interest in the internship opportunity with Ersilia, particularly in the Hugging Face Model Evaluation project. My background in machine learning, deep learning, and Python programming, along with my previous project experiences, make me an ideal candidate for this role.

During my previous projects, I have demonstrated my ability to work with state-of-the-art deep learning frameworks such as TensorFlow, PyTorch, and Hugging Face's Transformers library. In my recent role, I conducted a comprehensive benchmark of Hugging Face Generative Pre-trained Transformer models, evaluating their performance in generating text responses. This project not only honed my technical skills but also allowed me to develop a deep understanding of model evaluation methodologies and metrics.

Additionally, my experience in developing machine learning algorithms, such as the Grammar Error Corrector and the Brain Controlled Interface for Controlling Robotic Arm, has equipped me with a solid foundation in neural network architectures, attention mechanisms, and signal processing techniques. These skills will be invaluable in contributing to Ersilia's projects, especially in implementing advanced algorithms and models for natural language processing tasks.

Furthermore, I am deeply motivated by the opportunity to work on projects that have a meaningful impact on society. The Brain Controlled Interface project, in particular, allowed me to witness firsthand the transformative power of technology in improving the quality of life for individuals with disabilities. I am excited about the prospect of contributing to Ersilia's mission of developing innovative solutions that address real-world challenges.

Participating in the internship with Ersilia will not only advance my technical skills but also provide me with valuable industry experience and exposure to cutting-edge research and development practices. I am eager to collaborate with the talented team at Ersilia and contribute my expertise to meaningful projects.

Thank you for considering my application. I am enthusiastic about the opportunity to further discuss how my skills and experiences align with the goals of Ersilia. I am looking forward to the possibility of contributing to your team.

Sincerely,
Saisri Vishwanath

**Comment by saisri0102** [comment]
Week2 Tasks are uploaded to the below github repo:
https://github.com/saisri0102/model-validation/tree/main/model-validation

**Comment by saisri0102** [comment]
**WEEK 2: Get Familiar with Machine Learning for Chemistry**

Model Selected
Prediction of hERG Channel Blockers with DMPNN - eos30f3

Model Eos30f3 Description.
In drug discovery and medicine, there exists a component called the hERG potassium ion channel, responsible for regulating the flow of potassium ions essential for maintaining the heart's electrical activity. Certain substances can block this channel, leading to a condition known as hERG-mediated cardiotoxicity. A neural network model known as ChemProp, specifically the D-MPNN variant, has been developed to predict the cardiotoxicity potential of compounds by assessing their interaction with the hERG channel. This model was trained on a dataset comprising 7,889 molecules, with a concentration threshold of 10 uM.

Task 1 - Assessing Model Eos30f3 Bias.
The 1,000 Molecule Datasets Used in the Bias task were downloaded from ChEMBL. The code can be found in the below github repo: https://github.com/saisri0102/model-validation/tree/main/model-validation


Task 2 - Model Eos30f3 Reproducibility

Identify Results you want to reproduce

According to the publication, diverse classification models were trained using a neural network called directed message passing neural network (D-MPNN) on various datasets collected from multiple sources in order to identify compounds that inhibit hERG. The model that performed the best was the D-MPNN + moe206, achieving an AUC-ROC value of 0.956 ± 0.005. However, it's worth noting that the molecular descriptor moe206, utilized in this model, is proprietary, so the model implemented on Ersilia was trained without a molecule featurizer. We aim to replicate the original D-MPNN model, which achieved an AUC-ROC value of 0.947 ± 0.005, using a 5-fold cross validation with random splitting. This model was trained on a dataset comprising 7889 compounds with well-defined experimental data on hERG, encompassing diverse chemical structures and featuring 6 thresholds (10 μM, 20 μM, 40 μM, 60 μM, 80 μM, and 100 μM) for distinguishing hERG blockers from non-blockers. The author selected a 10 μM threshold for the model. This dataset was curated by Cai et al. and published in J Chem Inf Model, 2019.

Implement the model on your system as described by the authors
I cloned the model repository into my Ubuntu 22.4 system using git clone https://github.com/AI-amateur/DMPNN-hERG.git
The code can be found in the below github repo: https://github.com/saisri0102/model-validation/tree/main/model-validation

**Comment by DhanshreeA** [comment]
Hi @saisri0102 good work so far! Glad to see Ersilia Compound Embeddings being used for 2D visualizations. Could you do the following and then move onto the final application?:
1. Summarize in this issue, in a table, results from Task 1? 
2. Same as above, summarize reproducibility results for Task 2, including from your implementation of the paper, the EMH implementation, as well as the results published in the paper. Thank you.

I will review finally on Monday.

**Comment by saisri0102** [comment]
Week 2 Results:

Task 1:
<img width="1174" alt="Screenshot 2024-03-23 at 12 23 55 PM" src="https://github.com/ersilia-os/ersilia/assets/71775244/2dbcf3b1-6354-4c03-a3a4-01de130d3a1d">

Below are the distribution plots:
<img width="1512" alt="Screenshot 2024-03-23 at 12 26 40 PM" src="https://github.com/ersilia-os/ersilia/assets/71775244/0c5c3fb9-a7d4-41db-af32-a3f603d09a78">

<img width="1512" alt="Screenshot 2024-03-23 at 12 26 50 PM" src="https://github.com/ersilia-os/ersilia/assets/71775244/35a5c6d2-0198-419e-866c-528aa6f06891">
<img width="1512" alt="Screenshot 2024-03-23 at 12 26 55 PM" src="https://github.com/ersilia-os/ersilia/assets/71775244/ca5b8fb5-dea4-4dde-8314-337f1afc5291">

**Comment by saisri0102** [comment]
Week 2 Task 2 Results:

Predictions generated by the D-MPNN+moe206 model for the test dataset during the first run of a 5-fold cross-validation:
                                                smiles         class
0      C[C@@H]1CCC[NH+]1CCc2oc3ccc(cc3c2)c4cncc(c4)C#N  9.947536e-01
1         CCCCCCC[N@@H+](CC)CCCCc1ccc(cc1)[N+](=O)[O-]  9.365226e-01
2    CS(=O)(=O)Nc1ccc2OC3(CC[NH+](CC3)C4CCc5cc(ccc5...  9.982115e-01
3    Fc1ccc(cc1)n2cc(C3CC[NH+](CCN4CCNC4=S)CC3)c5cc...  9.781892e-01
4    Cc1ccc2c(cccc2n1)c3nnc(SCCC[NH+]4CCc5cc6[NH2+]...  9.908385e-01
..                                                 ...           ...
250  OC[C@H]1CCC[NH+]1CCCOc2ccc3c(Nc4cnn(CC(=O)Nc5c...  6.717938e-02
251  Clc1ccc2c(c1)c(cn2C3CCCCC3)C4CC[N+](CCN5CCNC5=...  9.655933e-01
252  CC[C@H]1OC(=O)[C@H](C)[C@@H](O[C@H]2C[C@@](C)(...  5.475458e-10
253  CN[C@@H](CC#N)[C@@H]1CCN(C1)c2c(F)cc3C(=O)C(=C...  4.197311e-04
254    COC(=O)[C@@H]1[C@@H](O)C[C@@H]2CC[C@H]1[N@H+]2C  6.115026e-01

Evaluation results of the D-MPNN+moe206 model:
<img width="1431" alt="image" src="https://github.com/ersilia-os/ersilia/assets/71775244/137d2ad8-692f-4919-91ce-e03528427f97">

Below are the predictions obtained from the Ersilia Model Hub implementation:

key | input | activity
-- | -- | --
UGELZTGBPPXJPE-OAHLLOKOSA-O | C[C@@H]1CCC[NH+]1CCc2oc3ccc(cc3c2)c4cncc(c4)C#N | 0.886925
YTYATOMQOOFRNA-UHFFFAOYSA-O | CCCCCCC[N@@H+](CC)CCCCc1ccc(cc1)[N+](=O)[O-] | 0.785938
NIYGLRKUBPNXQS-UHFFFAOYSA-O | CS(=O)(=O)Nc1ccc2OC3(CC[NH+](CC3)C4CCc5cc(ccc5... | 0.850671
UDRWVFGKMDCPTL-UHFFFAOYSA-O | Fc1ccc(cc1)n2cc(C3CC[NH+](CCN4CCNC4=S)CC3)c5cc... | 0.886333
JQEQULOLEPTBRS-UHFFFAOYSA-P | Cc1ccc2c(cccc2n1)c3nnc(SCCC[NH+]4CCc5cc6[NH2+]... | 0.774152
... | ... | ...
BQSSHYNPAGSOBT-LJQANCHMSA-O | OC[C@H]1CCC[NH+]1CCCOc2ccc3c(Nc4cnn(CC(=O)Nc5c... | 0.845857
CLQSZXFVERHVFU-UHFFFAOYSA-N | Clc1ccc2c(c1)c(cn2C3CCCCC3)C4CC[N+](CCN5CCNC5=... | 0.875263
PJVYTFDJHSYNLB-QNPWSHAKSA-N | CC[C@H]1OC(=O)[C@H](C)[C@@H](O[C@H]2C[C@@](C)(... | 0.208751
DCRAPCRZDJGSOF-PXAZEXFGSA-M | CN[C@@H](CC#N)[C@@H]1CCN(C1)c2c(F)cc3C(=O)C(=C... | 0.390227
QIQNNBXHAYSQRY-KZVJFYERSA-O | COC(=O)[C@@H]1[C@@H](O)C[C@@H]2CC[C@H]1[N@H+]2C | 0.198635

Below are the author predicted values:
                                             smiles         class
0      C[C@@H]1CCC[NH+]1CCc2oc3ccc(cc3c2)c4cncc(c4)C#N  9.947536e-01
1         CCCCCCC[N@@H+](CC)CCCCc1ccc(cc1)[N+](=O)[O-]  9.365226e-01
2    CS(=O)(=O)Nc1ccc2OC3(CC[NH+](CC3)C4CCc5cc(ccc5...  9.982115e-01
3    Fc1ccc(cc1)n2cc(C3CC[NH+](CCN4CCNC4=S)CC3)c5cc...  9.781892e-01
4    Cc1ccc2c(cccc2n1)c3nnc(SCCC[NH+]4CCc5cc6[NH2+]...  9.908385e-01
..                                                 ...           ...
250  OC[C@H]1CCC[NH+]1CCCOc2ccc3c(Nc4cnn(CC(=O)Nc5c...  6.717941e-02
251  Clc1ccc2c(c1)c(cn2C3CCCCC3)C4CC[N+](CCN5CCNC5=...  9.655933e-01
252  CC[C@H]1OC(=O)[C@H](C)[C@@H](O[C@H]2C[C@@](C)(...  5.475480e-10
253  CN[C@@H](CC#N)[C@@H]1CCN(C1)c2c(F)cc3C(=O)C(=C...  4.197315e-04
254    COC(=O)[C@@H]1[C@@H](O)C[C@@H]2CC[C@H]1[N@H+]2C  6.115029e-01

Evaluation results for the author implementaion:
<img width="1431" alt="image" src="https://github.com/ersilia-os/ersilia/assets/71775244/6e10f0fe-7f70-4e93-8900-62c20df469e3">

**Comment by saisri0102** [comment]
**Week 2 Task 2 Results:**

Predictions generated by the **D-MPNN** model for the test dataset during the first run of a 5-fold cross-validation:
smiles	class
C[C@@H]1CCC[NH+]1CCc2oc3ccc(cc3c2)c4cncc(c4)C#N	0.985650
CCCCCCC[N@@H+](https://chat.openai.com/c/CC)CCCCc1ccc(cc1)[N+](https://chat.openai.com/c/=O)[O-]	0.086860
CS(=O)(=O)Nc1ccc2OC3(CC[NH+](https://chat.openai.com/c/CC3)C4CCc5cc(ccc5)C4)OC2c1	0.995024
Fc1ccc(cc1)n2cc(C3CC[NH+](https://chat.openai.com/c/CCN4CCNC4=S)CC3)c5ccncn5	0.979510
Cc1ccc2c(cccc2n1)c3nnc(SCCC[NH+]4CCc5cc6[NH2+]CCC6c5cc4C(F)(F)F)s3	0.975559
...	...
OC[C@H]1CCC[NH+]1CCCOc2ccc3c(Nc4cnn(CC(=O)Nc5ccccc5)c4=O)c[nH]c3c2	0.746229
Clc1ccc2c(c1)c(cn2C3CCCCC3)C4CCN+CC4	0.921453
CC[[C@H](https://chat.openai.com/c/C)]1OC(=O)C@HC@@HOC1=O	0.000008
CN[C@@H](https://chat.openai.com/c/CC#N)[[C@@H](https://chat.openai.com/c/O)]1CCN(C1)c2c(F)cc3C(=O)C(=CC(=O)c23)C(F)(F)F	0.012800
COC(=O)[C@@H]1C@@HC[C@@H]2CC[C@H]1[N@H+]2C	0.448671


Evaluation results of the D-MPNN model:

col_names | roc | prc | Recall | Precision | f1 | BA | accuracy | TN | FP | FN | TP | SP | SE | NPV | MCC | cohen_kappa
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
class_class | 0.9552784077 | 0.9369958198 | 0.9259259259 | 0.8403361345 | 0.8810572687 | 0.8983371126 | 0.8941176471 | 128 | 19 | 8 | 100 | 0.8707482993 | 0.9259259259 | 0.9411764706 | 0.7890569999 | 0.7860538827
col_names | roc | prc | Recall | Precision | f1 | BA | accuracy | TN | FP | FN | TP | SP | SE | NPV | MCC | cohen_kappa
class_class | 0.9520030234 | 0.9317286147 | 0.8981481481 | 0.8584070796 | 0.8778280543 | 0.8946523054 | 0.8941176471 | 131 | 16 | 11 | 97 | 0.8911564626 | 0.8981481481 | 0.9225352113 | 0.7851123174 | 0.7844868063
col_names | roc | prc | Recall | Precision | f1 | BA | accuracy | TN | FP | FN | TP | SP | SE | NPV | MCC | cohen_kappa
class_class | 0.9555303603 | 0.9344225018 | 0.8611111111 | 0.8942307692 | 0.8773584906 | 0.8931405896 | 0.8980392157 | 136 | 11 | 15 | 93 | 0.925170068 | 0.8611111111 | 0.9006622517 | 0.7905753739 | 0.7901633118
col_names | roc | prc | Recall | Precision | f1 | BA | accuracy | TN | FP | FN | TP | SP | SE | NPV | MCC | cohen_kappa
class_class | 0.9642227261 | 0.9510559147 | 0.8425925926 | 0.91 | 0.875 | 0.8906840514 | 0.8980392157 | 138 | 9 | 17 | 91 | 0.9387755102 | 0.8425925926 | 0.8903225806 | 0.7907885536 | 0.7891221374
col_names | roc | prc | Recall | Precision | f1 | BA | accuracy | TN | FP | FN | TP | SP | SE | NPV | MCC | cohen_kappa
class_class | 0.9503653313 | 0.9274934124 | 0.8796296296 | 0.8796296296 | 0.8796296296 | 0.8955971277 | 0.8980392157 | 134 | 13 | 13 | 95 | 0.9115646259 | 0.8796296296 | 0.9115646259 | 0.7911942555 | 0.7911942555

**Comment by saisri0102** [comment]
@DhanshreeA I have summarised the results from week1 and week2. Can you please confirm if I can go ahead and submit my final application?

**Comment by DhanshreeA** [comment]
Hi @saisri0102 looks great! Thanks for your efforts. Please go ahead and submit the final application.
